src/java/org/apache/cassandra/index/sai/plan/QueryController.java

```diff
@@ -904,23 +904,21 @@ public int estimateAnnNodesVisited(Orderer ordering, int limit, long candidates)
     {
         Preconditions.checkArgument(limit > 0, "limit must be > 0");
 
-        IndexContext context = ordering.context;
-        Collection<MemtableIndex> memtables = context.getLiveMemtables().values();
-        View queryView = context.getView();
+        var queryView = queryContext.view;
```

Comment:
Add assert it is not null. The way of setting the view looks fragile.

Reply:
It is fragile. I'm not satisfied with the current design.

Reply:
I added the assertion.